sdk/core/Azure.Core/api/Azure.Core.netstandard2.0.cs

```diff
@@ -159,6 +159,13 @@ public partial struct AccessToken
         public override bool Equals(object? obj) { throw null; }
         public override int GetHashCode() { throw null; }
     }
+    public partial class AzureKeyCredential
```

Comment:
Depending on how general purpose you want to make the class, some services (although I'm not sure if any Azure services that do this) have multiple different API keys, any one of which could be used. In practical terms, it means that depending on which kind of API key is provided, different headers are used (e.g. an AppId API key is sent in the X-APPID-KEY header, and a user API key is sent in the X-USER-KEY header). 

I guess you could just subclass the key in a service specific key credential at that point, though.

Reply:
Correct me if I am wrong, but this scenarios is covered with the policy, as it currently allows callers to set the header of the key they are going to use: https://github.com/Azure/azure-sdk-for-net/pull/10668/files#diff-60cd6c136a6b3ad6e0dcfd38c41cb72fR18

Reply:
Maybe. The question would be how a user of one of our client libraries indicate that the specific ApiKeyCredential they have is a user api key vs. app api key in the example above. If there is only one type of api key, the policy can have that knowledge baked in. If there are two api key types (each with it's own corresponding header or query parameter) for the same service, someone is going to have to help the policy figure out which header to use.

Reply:
Thanks Johan! Yeah, definitely something to consider.
For now I am going to leave it as is, and will keep evaluating for those future scenarios